components/kyma-metrics-collector/pkg/keb/client.go

```diff
@@ -79,68 +76,78 @@ func (c Client) GetAllRuntimes(req *http.Request) (*kebruntime.RuntimesPage, err
 }
 
 func (c Client) getRuntimesPerPage(req *http.Request, pageNum int) (*kebruntime.RuntimesPage, error) {
+	var resp *http.Response
+	var err error
+	defer func() {
+		if resp != nil && resp.Body != nil {
+			err := resp.Body.Close()
```

Comment:
same as the previous situation:
do the request, get the response, check for error, defer the body close

Reply:
refactored.